darts/tests/models/forecasting/test_encoders.py

```diff
@@ -292,6 +312,284 @@ def helper_encoder_from_model(
+                # are dropped internally and appended again at the end
+                covs_inf2 = encoder.encode_inference(
+                    n=n, target=ts, covariate=covs_train, merge_covariate=merge_covs
+                )
+                # We get the same results with `merge_covariate=True` (as original covariates will still be in input)
+                if merge_covs:
+                    assert covs_inf2 == covs_inf
+                # We get only the minimum required time spans with `merge_covariate=False` as input covariates will
+                # are not in output of `encode_train()/inference()`
+                else:
+                    if isinstance(encoder.index_generator, PastCovariateIndexGenerator):
+                        assert len(covs_inf2) == input_chunk_length
+                        assert covs_inf2.end_time() == ts.end_time()
+                    else:
+                        assert (
+                            len(covs_inf2) == input_chunk_length + output_chunk_length
+                        )
+                        assert (
+                            covs_inf2.end_time()
+                            == ts.end_time() + ts.freq * output_chunk_length
+                        )
+
+                # we can use the output of `encode_inference()` as input for `encode_inference()` and get the
+                # same results (encoded components get overwritten)
+                covs_inf3 = encoder.encode_inference(
+                    n=1, target=ts, covariate=covs_inf2, merge_covariate=merge_covs
+                )
+                assert covs_inf3 == covs_inf2
+
+            test_routine(copy.deepcopy(enc), merge_covs=False)
+            test_routine(copy.deepcopy(enc), merge_covs=True)
+
+    def test_sequential_encoder_general(self):
+        ts = tg.linear_timeseries(length=24, freq="MS")
+        covs = tg.linear_timeseries(length=24, freq="MS")
+
+        input_chunk_length = 12
+        output_chunk_length = 6
+        add_encoders = {
+            "cyclic": {"past": ["month", "day"], "future": ["day", "month"]},
+            "datetime_attribute": {
+                "past": ["month", "year"],
+                "future": ["year", "month"],
+            },
+            "position": {
+                "past": ["absolute", "relative"],
+                "future": ["relative", "absolute"],
+            },
+            "custom": {
+                "past": [lambda idx: idx.month, lambda idx: idx.year],
+                "future": [lambda idx: idx.month, lambda idx: idx.year],
+            },
+            "transformer": Scaler(),
+        }
+        # given `add_encoders` dict, we expect encoders to generate the following components
+        comps_expected_past = pd.Index(
+            [
+                "month_sin",
+                "month_cos",
+                "day_sin",
+                "day_cos",
+                "month",
```

Comment:
Question: with this new components-checking logic, will the encoders avoid generating a component if there's already a component named `month` in the original series? If yes, could be worth maybe renaming the encoded components with something not likely to collide - e.g. `darts_enc_month` or something.

Reply:
Yep that makes sense, thanks

Reply:
With this implementation it's actually a bit different: if `month` is already in the original series, we remove it, generate `month` encodings from scratch, and add it back to the original series.

In a first approach I implemented it like you described. But in the single TS `fit()` case, we save the covariates along with the encodings to the model. Then when we call `predict(n=10)`, the model uses the saved covariates to generate the encodings -> the encodings would not be regenerated as already part of input covariates -> we would not have all required timestamps for prediction (as `encode_inference()` is basically bypassed).
We could handle this by removing the encodings from the saved covariates in case they were loaded, but i think it's not the cleanest way to do. WDYT @hrzn?

Reply:
Indeed it doesn't sound very clean. Let's leave it like that then. It might be good to document this in a docstring somewhere though, as some users will probably wonder how such cases are handled.